### ISSUE TITLE
Coverage is not reported correctly anymore

### ISSUE THREAD

**Issue by wilzbach** [storyscript/runtime#687]
Local execution yields this:

```
Name                                                    Stmts   Miss Branch BrPart  Cover   Missing
---------------------------------------------------------------------------------------------------
storyruntime/App.py                                       173      4     46      1    98%   307-308, 311-312, 208->206
storyruntime/AppConfig.py                                  18      0      2      0   100%
storyruntime/Apps.py                                      205      3     65      3    98%   348-353, 367, 347->348, 362->367, 405->exit
storyruntime/Config.py                                     17      0      2      0   100%
storyruntime/Containers.py                                141      0     43      4    98%   50->53, 54->60, 82->exit, 102->93
storyruntime/DeploymentLock.py                             13      0      2      0   100%
storyruntime/Exceptions.py                                 84      8     14      0    88%   123-127, 141-145
storyruntime/GraphQLAPI.py                                 43      7      8      1    84%   133-136, 139-144, 138->139
storyruntime/Kubernetes.py                                289      6    104     11    95%   178, 420, 443, 542-545, 174->180, 177->178, 369->365, 377->375, 419->420, 442->443, 539->550, 541->542, 631->634, 634->639, 646->657
storyruntime/Logger.py                                     63      3     14      2    94%   26, 35-36, 25->26, 34->35
storyruntime/Metrics.py                                     7      0      0      0   100%
storyruntime/Service.py                                    79      2      2      1    96%   39, 71, 70->71
storyruntime/ServiceUsage.py                               78      6     24      1    91%   88, 116-124, 167-169, 87->88
storyruntime/Story.py                                     160      2     66      5    97%   125, 220, 118->exit, 124->125, 208->222, 217->220, 305->304
storyruntime/Types.py                                       9      0      0      0   100%
storyruntime/Version.py                                     1      0      0      0   100%
storyruntime/__init__.py                                    0      0      0      0   100%
storyruntime/constants/ContextConstants.py                  6      0      0      0   100%
storyruntime/constants/Events.py                            7      0      0      0   100%
storyruntime/constants/LineConstants.py                     7      0      0      0   100%
storyruntime/constants/LineSentinels.py                    17      0      0      0   100%
storyruntime/constants/ReleaseConstants.py                  2      0      0      0   100%
storyruntime/constants/ServiceConstants.py                  3      0      0      0   100%
storyruntime/constants/__init__.py                          2      0      0      0   100%
storyruntime/db/Database.py                                82      3     16      0    97%   228-237
storyruntime/db/__init__.py                                 0      0      0      0   100%
storyruntime/entities/ContainerConfig.py                    3      0      0      0   100%
storyruntime/entities/Multipart.py                          3      0      0      0   100%
storyruntime/entities/Release.py                            3      0      0      0   100%
storyruntime/entities/ReportingEvent.py                    27      0      0      0   100%
storyruntime/entities/Volume.py                             3      0      0      0   100%
storyruntime/entities/__init__.py                           0      0      0      0   100%
storyruntime/enums/AppEnvironment.py                        6      0      2      0   100%
storyruntime/enums/ReleaseState.py                         13      0      2      0   100%
storyruntime/enums/__init__.py                              0      0      0      0   100%
storyruntime/http_handlers/BaseHandler.py                  27      0      2      0   100%
storyruntime/http_handlers/StoryEventHandler.py            62      0     14      2    97%   72->78, 111->115
storyruntime/http_handlers/__init__.py                      2      0      0      0   100%
storyruntime/omg/Exceptions.py                             16      0      0      0   100%
storyruntime/omg/ServiceOutputValidator.py                 38      0     24      0   100%
storyruntime/omg/__init__.py                                0      0      0      0   100%
storyruntime/processing/Lexicon.py                        259      3    132      5    98%   211, 523, 525, 168->174, 210->211, 522->523, 524->525, 570->574
storyruntime/processing/Mutations.py                       33      0     12      0   100%
storyruntime/processing/Services.py                       377      6    162      6    98%   62, 66, 516, 535, 588-589, 60->62, 63->66, 275->270, 351->exit, 515->516, 534->535
storyruntime/processing/Stories.py                         38      0      8      0   100%
storyruntime/processing/__init__.py                         3      0      0      0   100%
storyruntime/processing/internal/Decorators.py             11      0      0      0   100%
storyruntime/processing/internal/File.py                  100     13     26      3    86%   102, 127, 136, 158-166, 198, 230, 256-257, 275-276, 101->102, 126->127, 135->136
storyruntime/processing/internal/Http.py                   41      4     14      1    91%   59-60, 72-73, 51->54
storyruntime/processing/internal/Json.py                    8      0      0      0   100%
storyruntime/processing/internal/Log.py                    11      0      0      0   100%
storyruntime/processing/internal/__init__.py                0      0      0      0   100%
storyruntime/processing/mutations/FloatMutations.py        47      0      6      0   100%
storyruntime/processing/mutations/IntegerMutations.py      11      0      0      0   100%
storyruntime/processing/mutations/ListMutations.py         64      0     10      0   100%
storyruntime/processing/mutations/MapMutations.py          36      3     10      0    93%   9, 26-27
storyruntime/processing/mutations/StringMutations.py       43      0     10      0   100%
storyruntime/processing/mutations/__init__.py               0      0      0      0   100%
storyruntime/reporting/Reporter.py                         38      0     12      2    96%   25->34, 34->exit
storyruntime/reporting/ReportingAgent.py                    4      1      0      0    75%   6
storyruntime/reporting/__init__.py                          0      0      0      0   100%
storyruntime/reporting/agents/CleverTapAgent.py            31      0      6      0   100%
storyruntime/reporting/agents/SentryAgent.py               22      0      4      1    96%   44->47
storyruntime/reporting/agents/__init__.py                   0      0      0      0   100%
storyruntime/utils/ConstDict.py                            15      0      0      0   100%
storyruntime/utils/Dict.py                                 22      0     12      0   100%
storyruntime/utils/HttpUtils.py                            42      1     14      1    96%   26, 25->26
storyruntime/utils/RegExpUtils.py                          12      0      8      0   100%
storyruntime/utils/Resolver.py                            177     18    106      7    89%   41, 70-71, 74-77, 109, 118-120, 220, 227, 232-235, 258, 40->41, 62->70, 108->109, 116->118, 214->220, 225->227, 257->258
storyruntime/utils/StringUtils.py                          10      0      2      0   100%
storyruntime/utils/TypeResolver.py                        117      1     70      1    99%   107, 106->107
storyruntime/utils/TypeUtils.py                            34      1     20      1    96%   41, 40->41
storyruntime/utils/__init__.py                              3      0      0      0   100%
---------------------------------------------------------------------------------------------------
TOTAL                                                    3318     95   1096     59    96%
Coverage XML written to file coverage.xml
```

Compare with:

![image](https://user-images.githubusercontent.com/4370550/67845824-73083400-fb00-11e9-98b4-9a3af5c1de68.png)

https://codecov.io/gh/storyscript/runtime/tree/c7da4b687b38b61351f858daea96d9648116ced7/storyruntime

**Comment by wilzbach** [comment]
This used to work:

![image](https://user-images.githubusercontent.com/4370550/67846037-d3977100-fb00-11e9-85f1-207a7a7d0130.png)

https://codecov.io/gh/storyscript/runtime/tree/ec6f361c6c04c394bdf48ff451191cfdf4bc6167

So this can be tracked back to:

PR: https://github.com/storyscript/runtime/pull/637
CodeCov: https://codecov.io/gh/storyscript/runtime/tree/dda5b054de9434fd370b3b610ed1f58046301887

**Comment by wilzbach** [comment]
The generated XML that's reported to CodeCov looks good:

```
<coverage branch-rate="0.9317" branches-covered="1023" branches-valid="1098" complexity="0" line-rate="0.9714" lines-covered="3225" lines-valid="3320" timestamp="1572427271739" version="4.5.4">
```

However, 

here's how the XML looks from Storyscript:

```xml
        <sources>
                <source>/home/circleci/project</source>
        </sources>
        <packages>
                <package branch-rate="0.5102" complexity="0" line-rate="0.8213" name="storyscript">
                        <classes>
                                <class branch-rate="0.7143" complexity="0" filename="storyscript/Api.py" line-rate="0.9014" name="Api.py">
```

and here's from `runtime`:

```xml
	<sources>
		<source>/home/circleci/repo/storyruntime</source>
	</sources>
	<packages>
		<package branch-rate="0.9056" complexity="0" line-rate="0.9703" name=".">
			<classes>
				<class branch-rate="0.9783" complexity="0" filename="App.py" line-rate="0.9769" name="App.py">
```

**Comment by wilzbach** [comment]
From an earlier build:
https://codecov.io/gh/storyscript/runtime/tree/bf0fc32f03380c66dfacecb12ada95d51ec8e7b7 the XML looked like:

```
               <package branch-rate="0.9046" complexity="0" line-rate="0.9703" name="storyruntime">
                        <classes>
                                <class branch-rate="0.9783" complexity="0" filename="storyruntime/App.py" line-rate="0.9774" name="App.py">
```

tl;dr: CodeCov needs `filename`s which are relative to the project root or if this is not possible a fixup path config could be used:

https://docs.codecov.io/docs/fixing-paths

**Comment by wilzbach** [comment]
Looks good again:

![image](https://user-images.githubusercontent.com/4370550/67851476-d72ff580-fb0a-11e9-9955-132c7c60622b.png)

https://codecov.io/gh/storyscript/runtime/tree/8d4b069d9bb06560394a96c61e2de880751b0a44
